ui/src/AudioRecorder.js

```diff
@@ -0,0 +1,36 @@
+export class AudioRecorder {
+
+    constructor() {
+        this.recordingUrl = "";
+        this.mediaRecorder = "";
+        this.recordedChunks = [];
+    }
+
+    startRecording() {
+        navigator.mediaDevices.getUserMedia({ audio: true, video: false })
+            .then(function(stream) {
+
+            this.mediaRecorder = new MediaRecorder(stream);
+            this.mediaRecorder.start();
+
+        });
+    }
+
+    stopRecording() {
+        this.mediaRecorder.stop();
+
+        this.mediaRecorder.ondataavailable = function(ev) {
+            this.recordedChunks.push(ev.data);
```

Comment:
Do the ondataavailable and onstop event handlers need to be defined in the stopRecording function every time? Or can they be defined when this.mediaRecorder is initialized on line 13?

If that's the case, you might want to move these out into an initMediaRecorder function and call it right after it's created.